experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -335,13 +335,13 @@ end
 ###############################################################################
 # demazures character formula
 function _demazure_operator(r::RootSpaceElem, w::WeightLatticeElem)
-  fl, index_of_r = is_simple_root_with_index(r)
-  @req fl "not a simple root"
+  fl, index_of_r = is_positive_root_with_index(r)
+  @req fl "r is not a positive root"
```

Comment:
```suggestion
```

This is unnecessary. Since this is checked when doing `refl = reflect(w, r)` and `index_of_r` is never used.

Reply:
This is needed here, as `reflect` is checks for roots, but the demazure operator is only defined for positive roots

Reply:
This is needed here, as `reflect` is checks for roots, but the demazure operator is only defined for positive roots. But you are right that this does not need the index here.
```suggestion
  @req is_positive_root(r) "r is not a positive root"
```